sound/soc/sof/topology.c

```diff
@@ -146,11 +148,20 @@ static int sof_keyword_dapm_event(struct snd_soc_dapm_widget *w,
 	dev_dbg(sdev->dev, "received event %d for widget %s\n",
 		event, w->name);
 
+	/* get runtime PCM params using widget's stream name */
+	spcm = snd_sof_find_spcm_name(sdev, swidget->widget->sname);
+
 	/* process events */
 	switch (event) {
 	case SND_SOC_DAPM_PRE_PMU:
+		if (spcm->stream[stream].d0i3_active) {
+			dev_dbg(sdev->dev, "KWD worked in suspended, no need to start it\n");
```

Comment:
Remove references to KWD, this is a generic capability that can be used for other usages.

Reply:
We don't need this DAPM event handler of other use case e.g. deep-buffer playback, do we?